services/app/src/components/views/quest/cardtemplates/combat/Actions.tsx

```diff
@@ -357,7 +357,7 @@ export const handleCombatTimerStop = remoteify(function handleCombatTimerStop(a:
   dispatch(audioSet({peakIntensity: 0}));
 
   a.node = a.node.clone();
-  const arng = seedrandom.alea(a.seed);
+  const arng = seedrandom.alea(a.seed + combat.roundCount);
```

Comment:
A regression in this stuff could affect multiplayer syncing - can you add a unit test that the seed is the same for a player that plays through multiple rounds, then syncs multiplayer?

Probably also worth a comment that this is added because the seed is tied to the ParserNode (which remains the same node during combat). 

Alternatively we could make this easier by creating a new node after every action in combat, but that's a much larger change.

Reply:
Wouldn't creating a new node have the same problem, since it's not a problem with the node but rather that we're passing the same starting seed to a fresh, deterministic random generator?

Am not familiar enough with the multiplayer code to write a test for that in a bounded amount of time :( Are seeds generated new for each node? If so, that's good - but if, say, it's the same seed for the whole session or quest, we're going to have a problem with the rolls being the same for the same round of combat, i.e. the first round of each fight would have the same rolls

Reply:
Every time a new node is constructed, its [context is updated](https://github.com/ExpeditionRPG/expedition/blob/master/shared/parse/Node.tsx#L40) and [a new seed is generated](https://github.com/ExpeditionRPG/expedition/blob/master/shared/parse/Context.tsx#L207) based on the old seed. It can be overridden (e.g. when [cloning a node](https://github.com/ExpeditionRPG/expedition/blob/master/shared/parse/Node.tsx#L58)), but typically every new node has a different seed than the previous node. RNG is done by doing something like `const rng = seedrandom.alea(seed)` at the start of rendering a card, then `rng()` whenever a new value is needed. So by creating a new node when rendering a new combat card, the rng automatically uses a new seed for the new card.

Reply:
So, I see that shared/Node.tsx generates a new seed on each card (which is tested), and there are tests to make sure that nodes with the same seed are handled deterministically, but I'm not seeing a test that MULTIPLAYER_SYNC handles seeds correctly / seeds persist. Perhaps that would be enough to feel comfortable with this? Since I'm not sure how I'd do that as a whole integration test in a non-fragile manner

Reply:
I think that's what I was asking for in the first comment :) the MULTIPLAYER_SYNC would need to sync through a combat to expose any weirdness in seed caused by this or other combat PRs.